web/modules/custom/portland/src/EventSubscriber/SearchApiEventSubscriber.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Drupal\portland\EventSubscriber;
+
+use Drupal\search_api_solr\Event\PreQueryEvent;
+use Drupal\search_api_solr\Event\PostCreateIndexDocumentsEvent;
+use Drupal\search_api_solr\Event\SearchApiSolrEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class SearchApiEventSubscriber implements EventSubscriberInterface {
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents(): array {
+    return [
+      SearchApiSolrEvents::POST_CREATE_INDEX_DOCUMENTS => 'solrPostCreateIndexDocuments',
+      SearchApiSolrEvents::PRE_QUERY => 'solrPreQuery',
+    ];
+  }
+
+  /**
+   * Our "aggregated sorting date" field is used for date-based boosting. This event
+   * will change the date field that is used depending on the content type. For instance, on events we want to use start date.
+   */
+  public function solrPostCreateIndexDocuments(PostCreateIndexDocumentsEvent $event): void {
+    $documents = $event->getSolariumDocuments();
+    foreach ($documents as $document) {
+      if ($document->ss_content_type === 'news') {
+        // When editor did not specify an Updated On date, use Published On date
+        if ($document->ds_field_updated_on == null) {
+          $document->ds_aggregated_sorting_date = $document->ds_field_published_on;
+        } else {
+          $document->ds_aggregated_sorting_date = $document->ds_field_updated_on;
+        }
+      } else if ($document->ss_content_type === 'event') {
+        $document->ds_aggregated_sorting_date = $document->ds_field_start_date;
+      }
+    }
+
+    $event->setSolariumDocuments($documents);
+  }
+
+  /**
+   * Alter Solr query to boost relevancy score based on aggregated_sorting_date.
+   */
+  public function solrPreQuery(PreQueryEvent $event): void {
```

Comment:
I haven't had time to debug. On the multi-dev, the "freshness" boost doesn't appear to work. When searching for "historic landmarks", I'd expect an event that's scheduled for today (7/22) to be displayed close to the top. But it didn't appear until the third page. Not sure if the formula needs to be tweaked further.

Reply:
D'oh. Good catch. I made a small code change to clean up the if statement before I pushed this PR, but had a logic bug where the query hook was always getting skipped. Please try your test again.